scripts/asr_language_modeling/ngram_lm/train_kenlm.py

```diff
@@ -54,95 +53,148 @@
 the required memory significantly, and the LM and its binary blob format require less storage space.
 """
 
-CHUNK_SIZE = 8192
-CHUNK_BUFFER_SIZE = 512
-
 
 def main():
+    file_path = os.path.split(os.path.realpath(__file__))[0]
     parser = argparse.ArgumentParser(
         description='Train an N-gram language model with KenLM to be used with beam search decoder of ASR models.'
     )
     parser.add_argument(
-        "--train_file",
+        "--train_path",
         required=True,
+        nargs="+",
         type=str,
-        help="Path to the training file, it can be a text file or JSON manifest",
+        help="Path to the training files, or whitespace separated files, or folder with files. Files can be a text file, JSON manifest or .json.gz",
     )
+
     parser.add_argument(
-        "--nemo_model_file",
+        "--tokenizer_model_file",
         required=True,
         type=str,
-        help="The path of the '.nemo' file of the ASR model or name of a pretrained model",
+        help="The path to '.model' file of the SentencePiece tokenizer, or '.nemo' file of the ASR model, or name of a pretrained NeMo model",
     )
     parser.add_argument(
         "--kenlm_model_file", required=True, type=str, help="The path to store the KenLM binary model file"
     )
     parser.add_argument("--ngram_length", required=True, type=int, help="The order of N-gram LM")
+    parser.add_argument(
+        "--ngram_prun",
+        default="0",
+        required=False,
+        nargs="+",
+        type=str,
+        help="Whitespace separated digits. Example: 0 0 1. See Pruning section on the https://kheafield.com/code/kenlm/estimation ",
+    )
     parser.add_argument("--kenlm_bin_path", required=True, type=str, help="The path to the bin folder of KenLM")
+    parser.add_argument(
+        "--cache_path", required=False, type=str, default=None, help="Cache path to save tokenized files"
```

Comment:
Hmm if it's none create a cache in tmpdir ? It's too slow otherwise

Reply:
This is to avoid text tokenization several times during multiple runs. But if you need to do it only ones then cache_path is None, script will write directly to stdin of KenLM